src/server.ts

```diff
@@ -182,83 +184,84 @@ wss.on('connection', async (ws: Server.Ws) => {
-    const data = getMessage(message);
+  const onMessage = async (message: string, data: Messages) => {
+    Object.assign(data, getMessage(message));
 
-    if (!data) {
-      const message = prepareErrorMessage(-1, 'Cannot parse the message', clientId);
+    const { command, id, params } = data;
+    let response: string;
 
-      logger.debug(`[${clientId}] Received invalid message`, message);
-      ws.send(message);
-      return;
-    }
     logger.debug(
-      `[${clientId}] RECV MSG ID ${data.id} "${data?.command}" with params: ${JSON.stringify(
-        data.params,
-      )}`,
+      `[${clientId}] RECV MSG ID ${id} "${command}" with params: ${JSON.stringify(params)}`,
     );
 
-    switch (data.command) {
+    switch (command) {
       case MESSAGES.GET_SERVER_INFO: {
-        const message = await getServerInfo(data.id, clientId);
+        response = await getServerInfo(id, clientId);
 
-        ws.send(message);
         break;
       }
 
       case MESSAGES.GET_TRANSACTION: {
-        const message = await getTransaction(data.id, clientId, data.params.txId);
+        assertParams(params, { txId: 'string' });
+        response = await getTransaction(id, clientId, params.txId);
 
-        ws.send(message);
         break;
       }
 
       case MESSAGES.GET_BLOCK: {
-        const message = await getBlock(data.id, clientId, data.params.hashOrNumber);
+        assertParams(params, { hashOrNumber: 'string | number' });
+        response = await getBlock(id, clientId, params.hashOrNumber);
 
-        ws.send(message);
         break;
       }
 
       case MESSAGES.GET_ACCOUNT_UTXO: {
-        const message = await getAccountUtxo(data.id, clientId, data.params.descriptor);
+        assertParams(params, { descriptor: 'string' });
+        response = await getAccountUtxo(id, clientId, params.descriptor);
 
-        ws.send(message);
         break;
       }
 
       case MESSAGES.ESTIMATE_FEE: {
-        const message = await estimateFee(data.id, clientId);
+        response = await estimateFee(id, clientId);
 
-        ws.send(message);
         break;
       }
 
       case MESSAGES.GET_ACCOUNT_INFO: {
-        const message = await getAccountInfo(
-          data.id,
+        assertParams(
+          params,
+          { descriptor: 'string', details: 'details' },
+          { page: 'number', pageSize: 'number' },
+        );
+        response = await getAccountInfo(
+          id,
           clientId,
-          data.params.descriptor,
-          data.params.details,
-          data.params.page,
-          data.params.pageSize,
+          params.descriptor,
+          params.details,
+          params.page,
+          params.pageSize,
         );
 
-        ws.send(message);
         break;
       }
 
       case MESSAGES.GET_BALANCE_HISTORY: {
-        const message = await getBalanceHistory(
-          data.id,
+        assertParams(params, {
+          descriptor: 'string',
+          groupBy: 'number',
```

Comment:
This is sadly potentially breaking change as the code worked even with `groupBy` parameter passed as a string, to avoid it we should continue to accept string values.

Probably it's the same with `from` and `to`.

Reply:
12963fc should solve this